### PR TITLE
Filter transaction before chunk production

### LINKED ISSUE #1182: Calculate transaction costs in TX verifier

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Fixes #1182

### CODE REVIEW

File: near/src/runtime.rs [hunk 1]
```diff
@@ -412,19 +411,46 @@ impl RuntimeAdapter for NightshadeRuntime {
...
+            &apply_state,
+            &transaction,
+        ) {
             debug!(target: "runtime", "Tx {:?} validation failed: {:?}", transaction, err);
             return Err(err);
         }
         Ok(ValidTransaction { transaction })
     }
 
+    fn filter_transactions(
```
Comment (bowenwang1996) [inline near/src/runtime.rs]: why exactly is block_index needed?

File: runtime/runtime/src/lib.rs [hunk 2]
```diff
@@ -205,30 +239,81 @@ impl Runtime {
...
+                            "Transaction receiver_id {:?} doesn't match the access key receiver_id {:?}",
+                            &transaction.receiver_id,
+                            &function_call_permission.receiver_id,
+                        ).into());
+                    }
+                    if !function_call_permission.method_names.is_empty() {
+                        if function_call_permission
+                            .method_names
+                            .iter()
+                            .find(|method_name| &function_call.method_name == *method_name)
```
Comment (bowenwang1996) [inline runtime/runtime/src/lib.rs]: You might want to use `any` instead.

File: near/src/runtime.rs (hunk 1 above)
Comment (evgenykuzyakov) [reply near/src/runtime.rs]: To charge rent

File: chain/client/src/client.rs [hunk 3]
```diff
@@ -1360,12 +1380,6 @@ impl ClientActor {
                     self.shards_mgr.insert_transaction(shard_id, valid_transaction);
                     NetworkClientResponses::ValidTx
                 } else {
-                    // TODO(MarX): Forward tx even if I am a validator.
```
Comment (SkidanovAlex) [inline chain/client/src/client.rs]: Keep the TODO?

**Pull request merged**